pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -241,7 +233,7 @@ func (c *Cluster) makeMonDaemonContainer(monConfig *monConfig) v1.Container {
 			[]string{
 				"--foreground",
 				"--public-addr", joinHostPort(monConfig.PublicIP, monConfig.Port),
-				// --public-bind-addr is set in the config file at init time
+				"--public-bind-addr", joinHostPort("$(ROOK_PRIVATE_IP)", monConfig.Port),
```

Comment:
ceph evaluates the env variable here when it evaluates the arg?

Reply:
I think the answer to your question is yes, but I want to make sure there isn't any confusion. I wouldn't say that "ceph" evaluates the env var; it is "kubernetes" which evaluates it. I did a sample test for myself where I changed the `ceph-mon` command to be `echo`, and the pod log showed that the value was replaced by the pod's IP address. It is worth noting that `describe pod` will still show the text `$(ROOK_PRIVATE_IP)`.

Reply:
Ok I would be curious to confirm on a test pod if kubernetes was evaluating the env vars here. I believe I have seen in the past where ceph evaluates env vars for the command line params, so that was my assumption.

Reply:
Regarding this, I think I have a way to verify, but it's a little annoying since k8s obfuscates secrets. I had to use docker on the node itself to see the expanded info.

`docker inspect <container_id>` of the container which is running `ceph-mon --foregr...` shows the cmd as below for my test cluster. Because the entry following `--public-bind-addr` is the IP address that I can verify matches my pod's IP address with `kubectl get pods -o wide`, I can assume that it is kubernetes evaluating the expression and not Ceph.
```
"Cmd": [
                "--foreground",
                "--public-addr",
                "172.24.164.150:6790",
                "--public-bind-addr",
                "172.16.6.67:6790",
                "--name",
                "mon.a",
                "--mon-data",
                "/var/lib/rook/mon-a/data"
            ],
```